app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
   belongs_to :hotel
+  has_many :guest_rooms
+  has_many :guests, through: :guest_rooms
+
+  def guest_count
+    guests.count
+  end
+
+  def room_guests
+    guests.pluck(:name)
```

Comment:
Remember we asked for distinct names though, so you'd want to be sure to put guests.distinct.pluck(:name) to ensure no dups.